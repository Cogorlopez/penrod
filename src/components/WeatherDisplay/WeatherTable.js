import React from 'react';
import { DataTable } from '@salesforce/design-system-react';
import toString from '../../functions/toString';

import Columns from './Columns';

const WeatherTable = props => {
  return (
    <div className="slds-m-top_small">
      <DataTable
        className="slds-m-top_small"
        items={toString(props.weather)}
        id="DataTableExample-1-default"
        fixedLayout
      >
        {Columns}
      </DataTable>
    </div>
  );
};

export default WeatherTable;
