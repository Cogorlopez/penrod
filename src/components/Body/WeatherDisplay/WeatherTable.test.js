import React from 'react';
import TestRenderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import WeatherTable from './WeatherTable';

configure({ adapter: new Adapter() });

var weather = [];
weather.push({
  id: '5263045',
  name: 'Milwaukee',
  cod: '200',
  temp: '31.28',
  pressure: '1007',
  humidity: '75',
  temp_min: '26.6',
  temp_max: '35.06'
});

it('renders correctly when there is a weather object passed to it', () => {
  const tbl = TestRenderer.create(shallow(<WeatherTable weather={weather} />));

  expect(tbl).toMatchSnapshot();
});
