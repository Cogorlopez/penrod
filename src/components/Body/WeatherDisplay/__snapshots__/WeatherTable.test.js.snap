// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is a weather object passed to it 1`] = `
<div
  className="slds-m-top_small"
>
  <table
    className="slds-table slds-table_fixed-layout slds-table_resizable-cols slds-table_bordered slds-m-top_small"
    id="DataTableExample-1-default"
    role="grid"
  >
    <thead>
      <tr
        className="slds-line-height_reset"
      >
        <th
          aria-label="City"
          aria-sort="none"
          className="slds-text-title_caps"
          scope="col"
          style={
            Object {
              "height": null,
              "lineHeight": null,
              "width": "6em",
            }
          }
        >
          <span
            className="slds-p-horizontal_x-small"
            style={
              Object {
                "display": "flex",
              }
            }
          >
            <span
              className="slds-truncate"
              title="City"
            >
              City
            </span>
          </span>
        </th>
        <th
          aria-label="Current Temperature (f)"
          aria-sort="none"
          className="slds-text-title_caps"
          scope="col"
          style={
            Object {
              "height": null,
              "lineHeight": null,
              "width": "5em",
            }
          }
        >
          <span
            className="slds-p-horizontal_x-small"
            style={
              Object {
                "display": "flex",
              }
            }
          >
            <span
              className="slds-truncate"
              title="Current Temperature (f)"
            >
              Current Temperature (f)
            </span>
          </span>
        </th>
        <th
          aria-label="Min Temperature (f)"
          aria-sort="none"
          className="slds-text-title_caps"
          scope="col"
          style={
            Object {
              "height": null,
              "lineHeight": null,
              "width": "5em",
            }
          }
        >
          <span
            className="slds-p-horizontal_x-small"
            style={
              Object {
                "display": "flex",
              }
            }
          >
            <span
              className="slds-truncate"
              title="Min Temperature (f)"
            >
              Min Temperature (f)
            </span>
          </span>
        </th>
        <th
          aria-label="Max Temperature (f)"
          aria-sort="none"
          className="slds-text-title_caps"
          scope="col"
          style={
            Object {
              "height": null,
              "lineHeight": null,
              "width": "5em",
            }
          }
        >
          <span
            className="slds-p-horizontal_x-small"
            style={
              Object {
                "display": "flex",
              }
            }
          >
            <span
              className="slds-truncate"
              title="Max Temperature (f)"
            >
              Max Temperature (f)
            </span>
          </span>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className=""
      >
        <td
          className={null}
          role="gridcell"
          style={
            Object {
              "width": "6em",
            }
          }
        >
          <div
            className="slds-truncate"
            title="Milwaukee"
          >
            Milwaukee
          </div>
        </td>
        <td
          className={null}
          role="gridcell"
          style={
            Object {
              "width": "5em",
            }
          }
        >
          <div
            className="slds-truncate"
            title="31.28"
          >
            31.28
          </div>
        </td>
        <td
          className={null}
          role="gridcell"
          style={
            Object {
              "width": "5em",
            }
          }
        >
          <div
            className="slds-truncate"
            title="26.6"
          >
            26.6
          </div>
        </td>
        <td
          className={null}
          role="gridcell"
          style={
            Object {
              "width": "5em",
            }
          }
        >
          <div
            className="slds-truncate"
            title="35.06"
          >
            35.06
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
